arkouda/dataframe.py

```diff
@@ -382,25 +382,83 @@ def _repr_html_(self):
     def _ipython_key_completions_(self):
         return self._columns
 
-    def drop(self, keys):
+    def _drop_column(self, keys):
         """
         Drop a column or columns from the dataframe, in-place.
 
-        Parameters
-        ----------
-        keys : str or list
-            The column(s) to be dropped.
+        keys : list
+            The labels to be dropped on the given axis
         """
-
-        if isinstance(keys, str):
-            keys = [keys]
         for key in keys:
             # Do not allow the user to drop the index column
             if key == 'index':
                 raise KeyError('The index column may be reset, but not dropped.')
 
             del self[key]
 
+    def _drop_row(self, keys):
+        """
+        Drop a row or rows from the dataframe, in-place.
+
+        keys : list
+            The indexes to be dropped on the given axis
+        """
+        idx_list = []
+        last_idx = -1
+        # sort to ensure we go in ascending order.
+        keys.sort()
+        for i, k in enumerate(keys):
+            if not isinstance(k, int):
+                raise TypeError("Index keys must be integers.")
+            pda = self.index[(last_idx+1):k:1]
+            if pda.size > 0:
+                idx_list.append(pda)
+            last_idx = k
+
+        pda = self.index[(last_idx+1):]
+        if pda.size > 0:
+            idx_list.append(pda)
+
+        idx_to_keep = concatenate(idx_list)
+        for key, val in self.items():
```

Comment:
i don't thing `val` is being used here, so we can just do `for key in self.keys():`
or it might have to be `self.data.keys()`

Reply:
Updated to use only the keys in the loop.